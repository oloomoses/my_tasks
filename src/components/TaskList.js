/* eslint-disable react/forbid-prop-types */
import React from 'react';
import propTypes from 'prop-types';
import Stats from './Stats';

const TaskList = ({ tasks, updateTask }) => {
  const toggleTaskComplete = (id) => {
    const updatedTask = tasks.map((task) => {
      if (id === task.id) {
        return {
          ...task,
          complete: !task.complete,
          completedTime: new Date().toLocaleTimeString(),
        };
      }
      return task;
    });

    updateTask(updatedTask);
  };

  return (
    <>
      <ul className="task-list">
        {
        tasks.map((task) => (

          <li className="task stack-small" key={task.id}>

            <div className="avatar">
              <img
                src={task.avatar}
                alt="A"
                className="avatar-img"
                onError={(e) => {
                  if (e.target.src !== 'CE6.png') {
                    e.target.src = 'CE6.png';
                  }
                }}
              />
            </div>

            <div className="task-td">
              <div className="task-title">{task.name}</div>
              <div className="task-description">{task.description}</div>
            </div>

            <div className="cb">
              {task.complete ? (
                <div className="completed-time">{task.completedTime}</div>
              ) : (
                <input id="todo-0" type="checkbox" defaultChecked={task.complete} onChange={() => toggleTaskComplete(task.id)} />
              )}

            </div>

          </li>
        ))
      }

      </ul>
      <hr />
      <Stats />
    </>
  );
};

TaskList.propTypes = {
  tasks: propTypes.array.isRequired,
  updateTask: propTypes.func.isRequired,
};

export default TaskList;
