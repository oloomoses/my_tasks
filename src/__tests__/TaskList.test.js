import { shallow } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import TaskList from '../components/TaskList';

let wrapper;
let tasks;
let updateTask;
beforeEach(() => {
  tasks = [];
  updateTask = () => ({
    id: '3', name: 'Task name 3', description: 'task name 3 description', avatar: '', complete: false, completedTime: '',
  });
  wrapper = shallow(
    <BrowserRouter><TaskList tasks={tasks} updateTask={updateTask} /></BrowserRouter>,
  );
});

test('has stats', () => {
  expect(wrapper.find('Stats')).toBeTruthy();
});

it('matches the snapshot', () => {
  const tree = renderer.create(
    <BrowserRouter><TaskList tasks={tasks} updateTask={updateTask} /></BrowserRouter>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
