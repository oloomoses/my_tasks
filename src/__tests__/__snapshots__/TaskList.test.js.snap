// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
Array [
  <ul
    className="task-list"
  />,
  <hr />,
  <div
    className="stats-title"
  >
    Stats
  </div>,
  <ul
    className="stats"
  >
    <li
      className="stat stack-small"
    >
      <div
        className="stat-value"
      >
        0
      </div>
      <div
        className="stat-description"
      >
        Total Tasks Closed
      </div>
    </li>
    <li
      className="stat stack-small"
    >
      <div
        className="stat-value"
      >
        0
      </div>
      <div
        className="stat-description"
      >
        Total Tasks Open
      </div>
    </li>
    <li
      className="stat stack-small"
    >
      <div
        className="stat-value"
      >
        NaN
        %
      </div>
      <div
        className="percentage-description"
      >
        Completion Rate
      </div>
    </li>
  </ul>,
]
`;
